src/hooks/useKanban.js

```diff
@@ -0,0 +1,301 @@
+          updatedTasks[taskIndex] = updatedTask;
+          await updateColumn(columnId, { tasks: updatedTasks });
+        }
+        return;
+      }
+
+      // Adiciona o voto
+      const updatedTask = {
+        ...task,
+        votes: (task.votes || 0) + 1,
+        votedBy: [...(task.votedBy || []), voterId]
+      };
+
+      const updatedTasks = [...column.tasks];
+      updatedTasks[taskIndex] = updatedTask;
+      await updateColumn(columnId, { tasks: updatedTasks });
+
+      // Salva o voto no localStorage para usuários não logados
+      if (!user) {
+        const votedTasks = JSON.parse(localStorage.getItem('votedTasks') || '[]');
+        votedTasks.push(taskId);
+        localStorage.setItem('votedTasks', JSON.stringify(votedTasks));
+      }
+    } catch (error) {
+      console.error('Erro ao votar na task:', error);
+    }
+  };
+
+  // Verifica se um usuário já votou em uma task
+  const hasVoted = (taskId) => {
+    try {
+      const voterId = user ? user.uid : getVisitorId();
+      const column = Object.values(columns).find(col => 
+        col.tasks?.some(task => task.id === taskId)
+      );
+      
+      if (!column) return false;
+      
+      const task = column.tasks.find(t => t.id === taskId);
+      return task?.votedBy?.includes(voterId) || false;
+    } catch (error) {
+      console.error('Erro ao verificar voto:', error);
+      return false;
+    }
+  };
+
+  // Gera um ID único para visitantes não logados
+  const getVisitorId = () => {
+    let visitorId = localStorage.getItem('visitorId');
+    if (!visitorId) {
+      visitorId = 'visitor_' + Math.random().toString(36).substr(2, 9);
+      localStorage.setItem('visitorId', visitorId);
+    }
+    return visitorId;
+  };
+
+  // Move uma task entre colunas
+  const moveTask = async (sourceColumnId, destinationColumnId, taskId, sourceIndex, destinationIndex) => {
+    if (!user) return;
+    
+    try {
+      const sourceColumn = columns[sourceColumnId];
+      const destinationColumn = columns[destinationColumnId];
+      
+      if (!sourceColumn || !destinationColumn) return;
+
+      const sourceTasks = [...(sourceColumn.tasks || [])];
+      const [movedTask] = sourceTasks.splice(sourceIndex, 1);
+
+      const destinationTasks = sourceColumnId === destinationColumnId 
+        ? sourceTasks 
+        : [...(destinationColumn.tasks || [])];
+
+      destinationTasks.splice(destinationIndex, 0, movedTask);
+
+      if (sourceColumnId === destinationColumnId) {
+        await updateColumn(sourceColumnId, { tasks: destinationTasks });
+      } else {
+        await Promise.all([
+          updateColumn(sourceColumnId, { tasks: sourceTasks }),
+          updateColumn(destinationColumnId, { tasks: destinationTasks })
+        ]);
+      }
+    } catch (error) {
+      console.error('Erro ao mover task:', error);
+    }
+  };
+
+  // Atualiza uma task existente
+  const updateTask = async (columnId, taskId, updates) => {
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const updateTask = async (columnId, taskId, updates) => {
  if (!columnId || !taskId || !updates || typeof updates !== 'object') {
    throw new Error('Invalid parameters provided to updateTask');
  }
```
Add input validation for the updates parameter to ensure only valid fields are being updated. This prevents potential data corruption and improves code reliability.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;